File: software/bindings/java/ExampleCallback.java
import com.tinkerforge.BrickletRotaryPoti;
import com.tinkerforge.IPConnection;

public class ExampleCallback {
	private static final String host = new String("localhost");
	private static final int port = 4223;
	private static final String UID = new String("ABC"); // Change to your UID
	
	// Note: To make the example code cleaner we do not handle exceptions. Exceptions you
	//       might normally want to catch are described in the commnents below
	public static void main(String args[]) throws Exception {
		// Create connection to brickd
		IPConnection ipcon = new IPConnection(host, port); // Can throw IOException

		BrickletRotaryPoti rp = new BrickletRotaryPoti(UID); // Create device object

		// Add device to ip connection
		ipcon.addDevice(rp); // Can throw IPConnection.TimeoutException
		// Don't use device before it is added to a connection
		

		// Set Period for position callback to 0.05s (50ms)
		// Note: The position callback is only called every second if the 
		//       position has changed since the last call!
		rp.setPositionCallbackPeriod(50);

		// Add and implement position listener (called if position changes)
		rp.addListener(new BrickletRotaryPoti.PositionListener() {
			public void position(short position) {
				System.out.println("Position: " + position);
			}
		});
		
		System.out.println("Press ctrl+c to exit");
		ipcon.joinThread();
	}
}

File: software/examples/delphi/ExampleCallback.pas
program ExampleCallback;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  SysUtils, IPConnection, BrickletRotaryPoti;

type
  TExample = class
  private
    ipcon: TIPConnection;
    rp: TBrickletRotaryPoti;
  public
    procedure PositionCB(const position: smallint);
    procedure Execute;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = '9Hh'; { Change to your UID }

var
  e: TExample;

{ Callback function for position callback (parameter has range 0-100) }
procedure TExample.PositionCB(const position: smallint);
begin
  WriteLn(Format('Position: %d', [position]));
end;

procedure TExample.Execute;
begin
  { Create IP connection to brickd }
  ipcon := TIPConnection.Create(HOST, PORT);

  { Create device object }
  rp := TBrickletRotaryPoti.Create(UID);

  { Add device to IP connection }
  ipcon.AddDevice(rp);
  { Don't use device before it is added to a connection }

  { Set Period for position callback to 0.05s (50ms)
    Note: The position callback is only called every 50ms if the
          position has changed since the last call! }
  rp.SetPositionCallbackPeriod(50);

  { Register position callback to procedure PositionCB }
  rp.OnPosition := {$ifdef FPC}@{$endif}PositionCB;

  WriteLn('Press key to exit');
  ReadLn;
  ipcon.Destroy;
end;

begin
  e := TExample.Create;
  e.Execute;
  e.Destroy;
end.
